test/main_workflow_test.rb

```diff
@@ -35,12 +39,18 @@ def mock_client(name, res)
       mock_client("inst_kickoff", :next)
       mock_client("inst_rpmcopy", :next)
 
-      cmd_success = { "exit" => 0 }
-      cmd_fail = { "exit" => 1 }
       allow(Yast::Update).to receive(:clean_backup)
       allow(Yast::Update).to receive(:create_backup)
       allow(Yast::Update).to receive(:restore_backup)
-      allow(Yast::SCR).to receive(:Execute).and_return(cmd_success, cmd_fail)
+      allow(Yast::SCR).to receive(:foo)
+
+      allow(Yast::SCR).to receive(:Execute).with(bash_path, /snapper .*list-configs/)
+        .and_return(cmd_success)
+      allow(Yast::SCR).to receive(:Execute).with(bash_path, /snapper create/).and_return(cmd_fail)
+      allow(Yast::Report).to receive(:Error).with(/Failed to create filesystem snapshot/)
```

Comment:
I'm not sure about the purpose of these lines (simulating a failure in snapper). Could you add a comment explaining?